lib/fakefs/base.rb

```diff
@@ -102,18 +118,19 @@ def without
       if !activated?
         yield
       else
+        io_mocked = @io_mocked
         begin
           deactivate!
           yield
         ensure
-          activate!
+          activate!(io_mocks: io_mocked)
```

Comment:
consistency would be nice here too
```suggestion
          activate!(io_mocked: io_mocked)
```

Reply:
hum are you sure of your suggestion? the keyword is `io_mocks` for all other cases.
I created a variable named `io_mocked` to keep the former state before the deactivate calls, I'm not sure this is enough to replace the name of the keyword.
I can rename the variable though :D